### PR TITLE
Handle non-delimited address lists

### PR DESCRIPTION
**Pull request by requaos**

### CODE REVIEW

**Comment by requaos** [comment]
I tried to follow the TravisCI output, made no sense, couldn't find the test it mentioned and the line number in `encode_test.go:179` is just `if err != nil {`... fishy
```✔ ~/go/src/github.com/jhillyerd/enmime [requaos/multipleFrom|⚑ 1] 
18:08 $ make
go clean github.com/jhillyerd/enmime github.com/jhillyerd/enmime/cmd github.com/jhillyerd/enmime/cmd/mime-dump github.com/jhillyerd/enmime/cmd/mime-extractor github.com/jhillyerd/enmime/internal/coding github.com/jhillyerd/enmime/internal/stringutil github.com/jhillyerd/enmime/internal/test
go test -race ./...
ok      github.com/jhillyerd/enmime     2.162s
ok      github.com/jhillyerd/enmime/cmd 1.583s
?       github.com/jhillyerd/enmime/cmd/mime-dump       [no test files]
?       github.com/jhillyerd/enmime/cmd/mime-extractor  [no test files]
ok      github.com/jhillyerd/enmime/internal/coding     1.566s
ok      github.com/jhillyerd/enmime/internal/stringutil 1.553s
ok      github.com/jhillyerd/enmime/internal/test       1.566s
go build
```

**Comment by jhillyerd** [comment]
Not your fault, I moved #102 from master to develop, so its golden test file does not reflect the newline changes from #99.  I will fix it tonight and re-run your test.

File: envelope_test.go [hunk 1]
```diff
@@ -869,6 +869,24 @@ func TestUnquotedSpecialCharParamsInMime(t *testing.T) {
 	}
 }
 
+func TestBadAddressHeaderInMime(t *testing.T) {
+	msg := test.OpenTestData("mail", "malformedMultipleAddressHeaderValues.raw")
```
Comment (jhillyerd) [inline envelope_test.go]:
```suggestion
	msg := test.OpenTestData("mail", "malformed-multiple-address-header-values.raw")
```
All the other testdata files are kabob-case.  Obv need to rename the actualy file.

Comment (requaos) [reply envelope_test.go]: done

**Comment by jhillyerd** [comment]
Thank you!

**Comment by requaos** [comment]
As a follow up to this, since I was neck deep in the RFC yesterday, do we enforce this rule during encoding:
If multiple addresses are present in the "From" address header, then a single address in the "Sender" address header must exist.

Conversely, as long as only one "From" address is present, then the "Sender" header is not required.

Should I open an issue? Since I have not even investigated if this logic is present (TLDR)

**Comment by jhillyerd** [comment]
Worth tracking in an issue.  I believe that the builder does not allow multiple From (or any Sender); and its build method is where most validation happens at this point, which is not much at all.

My feeling is if someone wants to build an email with Part objects manually, then they need to do their own validation.

**Pull request merged**